pilot/pkg/features/pilot.go

```diff
@@ -524,6 +524,9 @@ var (
 		true,
 		"If false, TCP probes will not be rewritten and therefor always succeed when a sidecar is used.",
 	).Get()
+
+	EnableInternalTrafficPolicy = env.RegisterBoolVar("PILOT_ENABLE_INTERNAL_TRAFFIC_POLICY", false,
+		"If true, pilot will support kubernetes service internal traffic policy (requires kubernetes version >= v1.21)").Get()
```

Comment:
edge case to consider - what happens with multinetwork? What if the multinetwork Gateway is on the same node with a pod? on a different node?

Reply:
> edge case to consider - what happens with multinetwork? What if the multinetwork Gateway is on the same node with a pod? on a different node?

k8s svc InternalTrafficPolicy is mainly internal, used inside the cluster. 
The gateway is for external access.

Reply:
Not for multinetwork - there is no reason that you cannot use InternalTrafficPolicy for a service spaning multiple networks.

Reply:
Just so I understand, k8s internal traffic policy can be either `cluster` (i.e. requests stay in the same cluster) or `local` (i.e. requests stay in the same node), right? So in either case, multi-network requests would not occur?

Reply:
Its:

Cluster - misnomer for "Everywhere"
Local - same node only

We cannot make "cluster" mean "same cluster" since its the default k8s option. I tried to get them to rename it to All but no luck

Reply:
Ah that rings a bell :)

So it would seem that multi-network wouldn't be an issue for "Local" since it wouldn't go to endpoints beyond the local node, right?

Reply:
Yes, assuming you don't have two networks on one node which I think is currently a reasonable assumption